core/trino-main/src/main/java/io/trino/operator/HttpPageBufferClient.java

```diff
@@ -485,6 +498,14 @@ public void onFailure(Throwable t)
         }, pageBufferClientCallbackExecutor);
     }
 
+    @VisibleForTesting
+    synchronized void updateAverageRequestSize(int successfulRequests, long responseSize)
+    {
+        if (successfulRequests > 0) {
+            averageRequestSizeInBytes = (averageRequestSizeInBytes * (successfulRequests - 1) + responseSize) / successfulRequests;
```

Comment:
in current could this is using doubles for computations:
```
            // AVG_n = AVG_(n-1) * (n-1)/n + VALUE_n / n
            averageBytesPerRequest = (long) (1.0 * averageBytesPerRequest * (successfulRequests - 1) / successfulRequests + responseSize / successfulRequests);
```
I think its more accurate

Reply:
right